### PR TITLE
Move available for purchase and visible in listings to product channel listing

### PR DESCRIPTION
**Pull request by fowczarek**
I want to merge this change because moving `available_for_purchase` and `visible_in_listings` to `ProductChannelListing`

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-902 Move available_for_purchase and visible_in_listings to ProductChannelListing](https://app.clickup.com/t/2549495/SALEOR-902)

**Comment by patrys** [comment]
@fowczarek Woah, that database impact 😨

**Comment by fowczarek** [comment]
@patrys Diff is between master and whole multichannel. We have tasks for data loaders on a multichannel list. Everything is under control. 
https://app.clickup.com/t/2549495/SALEOR-893

File: saleor/graphql/product/mutations/channels.py [hunk 1]
```diff
@@ -68,6 +85,29 @@ class Meta:
         error_type_class = ProductChannelListingError
         error_type_field = "product_channel_listing_errors"
 
+    @classmethod
+    def clean_available_for_purchase(cls, cleaned_input, errors: ErrorType):
+        channels_wit_invalid_available_for_purchase = []
```
Comment (IKarbowiak) [inline saleor/graphql/product/mutations/channels.py:90]: Typo:
```suggestion
        channels_with_invalid_available_for_purchase = []
```

File: saleor/graphql/product/mutations/channels.py [hunk 2]
```diff
@@ -90,10 +130,28 @@ def validate_product_without_category(cls, cleaned_input, errors: ErrorType):
...
+            defaults = {}
+            is_published = add_channel.get("is_published", None)
+            if is_published is not None:
+                defaults["is_published"] = is_published
+            publication_date = add_channel.get("publication_date", None)
+            if publication_date:
+                defaults["publication_date"] = publication_date
+            visible_in_listings = add_channel.get("visible_in_listings", None)
+            if visible_in_listings is not None:
+                defaults["visible_in_listings"] = visible_in_listings
```
Comment (IKarbowiak) [inline saleor/graphql/product/mutations/channels.py:142]: The same pattern is repeated 3 times, maybe we can use for loop here:
```
for field in ["is_published", "publication_date", "visible_in_listings"]:
            field_value = add_channel.get(field, None)
            if field_value is not None:
                defaults[field] = field_value
```

**Review by IKarbowiak — APPROVED** [review]

**Review by d-wysocki — APPROVED** [review]

**Pull request merged**